src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py

```diff
@@ -269,23 +272,44 @@ def histogram(
         else:
             data_range = None
 
-        for hist_type, target_object in zip(data_list, target_objects):
-            if len(hist_type) != 0:
-                hist, bins = histogram(hist_type, 20, data_range, density=False)
-            else:
-                hist = empty(0)
-                bins = empty(0)
-            quad = plot.quad(top=hist, bottom=0,
-                             left=bins[:-1], right=bins[1:],
-                             line_color='white', alpha=0.5,
-                             color=color_selector.get_color())
-            legend_manager.add_legend(target_object, quad)
-            hover = HoverTool(
-                tooltips=[(x_label, '@left'), ('The number of samples', '@top')], renderers=[quad]
-                )
-            plot.add_tools(hover)
+        hists = []
+
+        for hist_type in data_list:
+            hist, bins = histogram(hist_type, 20, data_range, density=False)
+            hists.append(hist)
+
+        hists_t = np.array(hists).T
+
+        colors = [
+            color.RGB(randint(0, 255), randint(0, 255), randint(0, 255))
+            for _ in range(len(hists_t))
+            ]
+
+        quad_dicts = {}
+
+        for i, h in enumerate(hists_t):
+            data = list(zip(h, colors, target_objects))
+            data.sort(key=lambda x: x[0], reverse=True)
+            for top, col, target_object in data:
+                quad = plot.quad(
```

Comment:
Is it possible to make the `legend` of a histogram with no data (all elements are 0) transparent?

If you can't come up with an idea, I can support you.